histomicstk/preprocessing/color_conversion/rgb_to_sda.py

```diff
@@ -0,0 +1,47 @@
+import numpy as np
+
+def rgb_to_sda(im_rgb, I_0, allow_negatives=False):
+    """Transform input RGB image or matrix `im_rgb` into SDA (stain
+    darkness) space for color deconvolution.
+
+    Parameters
+    ----------
+    im_rgb : array_like
+        Image (MxNx3) or matrix (3xN) of pixels
+
+    I_0 : float or array_like
+        Background intensity, either per-channel or for all channels
+
+    allow_negatives : bool
+        If True, would-be negative values in the output are clipped to 0
+
+    Returns
+    -------
+    im_sda : array_like
+        Shaped like `im_rgb`, with output values 0..255 where `im_rgb` >= 1
+
+    Note
+    ----
+    For compatibility purposes, passing I_0=None invokes the behavior of
+    rgb_to_od.
+
+    See Also
+    --------
+    histomicstk.preprocessing.color_conversion.sda_to_rgb,
+    histomicstk.preprocessing.color_conversion.rgb_to_od,
+    histomicstk.preprocessing.color_deconvolution.color_deconvolution,
+    histomicstk.preprocessing.color_deconvolution.color_convolution
+    """
+    is_matrix = im_rgb.ndim == 2
+    if is_matrix:
+        im_rgb = im_rgb.T
```

Comment:
Instead of this you can just add a check to ensure it is in the shape you want `3xN` or `Nx3` if not raise exception. Check for row or column size to be 3 depending on your preferred format.